app/controllers/SavedDeclarationsController.scala

```diff
@@ -40,7 +40,7 @@ class SavedDeclarationsController @Inject()(
 
   def displayDeclarations(pageNumber: Int = 1): Action[AnyContent] = authenticate.async { implicit request =>
     customsDeclareExportsConnector.findSavedDeclarations(Page(pageNumber, appConfig.paginationItemsPerPage)).map { page =>
-      Ok(savedDeclarationsPage(page))
+      Ok(savedDeclarationsPage(page)).removingFromSession(ExportsSessionKeys.declarationId)
```

Comment:
Why have this here?

Reply:
Why not?  Seemed like a good place to clear up session state.  We go from the list of saved decs to editing a dec (setting the dec id) landing on the summary page.  If we click "back"  we return to the saved dec list and back again takes us to the choice page with the dec type pre-selected.

Alternatively, we could clear the id when loading the choice page.   But then someone would complain when coming all the way back from doing a declaration.

Reply:
> cleared the declarationId from session when landing on the "saved drafts" page so that when you navigate back to the choice page the previously viewed dec type is not pre-selected.
Probably we should add a test for this functionality.

Reply:
Adding test... But are we happy that this is the right place to clear id?

Reply:
Any ideas how to test this?  I thought of creating a new controller test like this 
```
    "clear declarationId from session" when {
      "returning to display saved declarations" in new SetUp {
        getDeclaration("123")
        val editResult = controller.continueDeclaration("123")(getRequest())
        session(editResult).get(ExportsSessionKeys.declarationId) must be(Some("123"))

        listOfDraftDeclarations()
        val listResult = controller.displayDeclarations()(getRequest())
        session(listResult).get(ExportsSessionKeys.declarationId) must be(None)
      }
```
But that passes even without my change.  The second session is not related to the first.

Reply:
Ah it’s for back button behaviour 👍🏽 I’d prefer this kind of thing to be on the “exiting editing a declaration” actions e.g submit, save for later but I guess we need to handle back button logic somehow too